File: Models/Api/AuthModels.cs
using System.ComponentModel.DataAnnotations;

namespace SportsManagementMVC.Models.Api
{
    public class LoginRequest
    {
        [Required]
        [EmailAddress]
        [StringLength(254)]
        public string Email { get; set; } = string.Empty;

        [Required]
        [StringLength(128)]
        public string Password { get; set; } = string.Empty;
    }

    public class LoginResponse
    {
        public string Token { get; set; } = string.Empty;

        public DateTime ExpiresAt { get; set; }

        public AppUserResponse User { get; set; } = new();
    }

    public class AppUserResponse
    {
        public int Id { get; set; }

        public string Email { get; set; } = string.Empty;

        public string Role { get; set; } = string.Empty;

        public int? PlayerId { get; set; }

        public string? PlayerName { get; set; }
    }
}

File: mobile/app/src/main/java/com/paravolley/mobile/network/AuthModels.kt
package com.paravolley.mobile.network

data class LoginRequest(
    val email: String,
    val password: String
)

data class LoginResponse(
    val token: String,
    val expiresAt: String,
    val user: AppUserResponse
)

data class AppUserResponse(
    val id: Int,
    val email: String,
    val role: String,
    val playerId: Int?,
    val playerName: String?
)

data class RegisterPlayerRequest(
    val fullName: String,
    val email: String,
    val phone: String,
    val dateOfBirth: String,
    val province: String?,
    val town: String?,
    val experienceLevel: String?,
    val preferredPosition: String?,
    val classification: String,
    val emergencyContactName: String?,
    val emergencyContactPhone: String?,
    val medicalNotes: String?,
    val consent: Boolean
)

data class RegisterPlayerResponse(
    val message: String,
    val applicationId: Int
)
